app/src/main/kotlin/Reversi/ReversiSolver.kt

```diff
@@ -1,22 +1,48 @@
 package Reversi
 
+import StateNode
 import StatefulSolver
-import StatelessSolver
 import java.awt.Point
-import kotlin.random.Random
-
-class ReversiSolver(private val initialState: ReversiState) {
-    fun getMove(state: ReversiState) : Point {
-        val game = ReversiMDP(state)
-        val solver = StatelessSolver(
-            game,
-            2000,
-            1.4,
-            0.9,
-            true
-        )
-
-        solver.constructTree(60)
-        return solver.extractOptimalAction()
+
+class ReversiSolver(initialState: ReversiState)
+    : StatefulSolver<ReversiState, Point>(ReversiMDP(initialState), 2000, 1.4, 0.9, false) {
+
+    override fun iterateStep() {
+//        displayTree(4)
+        super.iterateStep()
```

Comment:
What is the point of this override? If its just calling `iterateStep()` of the super class?

Reply:
Mainly for printing the stuff earlier.